src/Rule/AbstractCompare.php

```diff
@@ -41,11 +43,28 @@ abstract class AbstractCompare implements
     use SkipOnErrorTrait;
     use WhenTrait;
 
+    /**
+     * A default for {@see $incorrectInputMessage}.
+     */
+    protected const DEFAULT_INCORRECT_INPUT_MESSAGE = 'The allowed types are integer, float, string, boolean and null.';
+    /**
+     * A default for {@see $incorrectDataSetTypeMessage}.
+     */
+    protected const DEFAULT_INCORRECT_DATA_SET_TYPE_MESSAGE = 'The attribute value returned from a custom data set ' .
+    'must have a scalar type.';
+
+    /**
+     * @var string[] List of valid types.
+     *
+     * @see CompareType
+     */
+    private static array $validTypes = [CompareType::STRING, CompareType::NUMBER];
```

Comment:
```suggestion
    private const VALID_TYPES = [CompareType::STRING, CompareType::NUMBER];
```

Reply:
Why was it marked as resolved?

Reply:
The first time the push was unsuccessful. Now it's OK.